tsmarty2c.php

```diff
@@ -145,7 +145,7 @@ function do_file($outfile, $file) {
 
 		if ($msg_ctxt && empty($result_msgctxt[$msg_ctxt][$msgid])) {
 			$result_msgctxt[$msg_ctxt][$msgid] = array();
-		} elseif (empty($result_msgid[$msgid])) {
+		} elseif (empty($msg_ctxt) && empty($result_msgid[$msgid])) {
```

Comment:
you don't need `empty()` around `$msg_ctxt`, it's simple variable that's already defined, so `!$msg_ctxt` should do the same